GitUI/GitUICommands.cs

```diff
@@ -356,6 +356,44 @@ public bool StartCheckoutRemoteBranch(IWin32Window owner, string branch)
 
         #endregion
 
+        /// <summary>
+        /// Spawns a new GE instance.
+        /// </summary>
+        /// <param name="arguments">The command line arguments.</param>
+        /// <param name="workingDir">The working directory for the new process.</param>
+        /// <returns>The <see cref="IProcess"/> object for controlling the spawned instance.</returns>
+        public static IProcess Spawn(string arguments, string workingDir = "")
+            => new Executable(Application.ExecutablePath, workingDir).Start(arguments);
+
+        /// <summary>
+        /// Spawns a new GE instance.
+        /// </summary>
+        /// <param name="workingDir">The working directory for the new process.</param>
+        /// <param name="selectedId">The optional commit to be selected.</param>
+        /// <param name="firstId">The first commit to be selected, the first commit in a diff.</param>
+        /// <returns>The <see cref="IProcess"/> object for controlling the spawned instance.</returns>
+        public static IProcess SpawnBrowse(string workingDir = "", ObjectId selectedId = null, ObjectId firstId = null)
+        {
+            StringBuilder arguments = new StringBuilder("browse");
```

Comment:
Consider using [`ArgumentBuilder`](https://github.com/gitextensions/gitextensions/blob/master/GitExtUtils/ArgumentBuilder.cs).

Reply:
Yes

Reply:
`ArgumentBuilder` inserts unwanted spaces

```
            var arguments = new ArgumentBuilder
            {
                "browse",
                { selectedId is not null, $"-commit={selectedId}" },
                { firstId is not null, $",{firstId}" }
            };
```

Reply:
The argument to -commit could be quoted or we add another argument `--firstCommit` too
(I prefer it like it is)
Should be written as:
```C#
var arguments = new ArgumentBuilder
            {
                "browse",
                { string.IsNullOrEmpty(selectedId), $"-commit={selectedId}" },
                { string.IsNullOrEmpty(selectedId) && string.IsNullOrEmpty(firstId), $",{firstId}" }
            };
```

Reply:
Can `ObjectId.ToString()` return an empty string for a non-null instance? 

RE first variant:
- It will insert a space in front of the comma.
- No need to check both `selectedId` and `firstId`. This handled in the code above. They are of type `ObjectId` not `string` here.

RE second variant:
- Neither the nested `$"{$"{}"}"` nor `$"{cond, optionalValue}"` do compile.
- Adding another instance of `ArgumentBuilder` will complicate the code more than the use of `StringBuilder`.

Reply:
> Can `ObjectId.ToString()` return an empty string for a non-null instance?

Not in the current implementation, so it is safe in the reality.
It is hard to explicitly assign null too, you have to try really hard: TryParse("xxx", out id)
So "is null" sufficient.

But the problem with the examples was that firstCommit must nut be included if secondCommit is not set
(The alternatives were not meant to be complete)
The PR is correct and what I prefer

Reply:
> `ArgumentBuilder` inserts unwanted spaces
> 
> ```
>             var arguments = new ArgumentBuilder
>             {
>                 "browse",
>                 { selectedId is not null, $"-commit={selectedId}" },
>                 { firstId is not null, $",{firstId}" }
>             };
> ```

Does it matter if the command is like `browse -commit=234235,`?

```cs
            var arguments = new ArgumentBuilder
            {
                "browse",
                { selectedId is not null, $"-commit={selectedId},{firstId}" }
            };
```

Reply:
> Does it matter if the command is like browse -commit=234235,?

I believe the parser must be tweaked